apps/server/src/shared/domain/service/permission.service.ts

```diff
@@ -53,10 +61,95 @@ export class PermissionService {
 		return hasPermissions;
 	}
 
-	checkUserHasAllSchoolPermissions(user: User, requiredPermissions: string[]): void {
-		const hasPermission = this.hasUserAllSchoolPermissions(user, requiredPermissions);
+	/**
+	 * Determines whether a user has all the rights specified in requiredPermissions
+	 * @param user
+	 * @param requiredPermissions
+	 * @throws UnauthorizedException
+	 */
+	checkUserHasAllPermissions(user: User, requiredPermissions: string[]): void {
+		const hasPermission = this.hasUserAllPermissions(user, requiredPermissions);
+		if (hasPermission !== true) {
+			throw new UnauthorizedException();
+		}
+	}
+
+	/**
+	 * Determines whether a user has at least one of the rights specified in requiredPermissions
+	 * @param user
+	 * @param requiredPermissions
+	 * @returns boolean
+	 */
+	hasUserOneOfPermissions(user: User, requiredPermissions: string[]): boolean {
+		if (!Array.isArray(requiredPermissions) || requiredPermissions.length === 0) {
+			return false;
+		}
+		const permissions = this.resolvePermissions(user);
+		const hasPermission = requiredPermissions.some((p) => permissions.includes(p));
+		return hasPermission;
+	}
+
+	/**
+	 * Determines whether a user has at least one of the rights specified in requiredPermissions
+	 * @param user
+	 * @param requiredPermissions
+	 * @throws UnauthorizedException
+	 */
+	checkUserHasOneOfPermissions(user: User, requiredPermissions: string[]): void {
+		const hasPermission = this.hasUserOneOfPermissions(user, requiredPermissions);
 		if (hasPermission !== true) {
 			throw new UnauthorizedException();
 		}
 	}
+
+	/**
+	 * Determines whether a user has access to the specified entity.
+	 * @example ```
+	 * const user = new User({id: 1})
+	 * const entity = new Course({id:2, creator: user})
+	 * const props = ['creator']
```

Comment:
In course we have no creator prob. :D 
I think we should optimize the params of the function that we do not need the explaination.
props => searchedEntityProps ?
entity => testedEntity ?

not well but it is more a ...userExistInTestedEntityKeys from what the method is do.